iOS/Resources/bin/arm64-apple-ios-ar

```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
+xcrun --sdk iphoneos ar $@
```

Comment:
The use of these stub scripts for executables is a clever solution. As they stand, I think there is one potential shortcoming if I understand the `xcrun` documentation correctly (I haven't yet verified this): according to the man page, specifying `--sdk` to `xcrun` overrides any `SDKROOT` environment variable, something which I've had occasion to use when building and testing for multiple macOS versions. It's probably less of a deal on iOS but I think it would be nice to allow for this. If the behavior is as documented, perhaps the stub scripts could do a test for the presence of an SDKROOT env variable and, if defined, not specify `--sdk iphoneos`?

Reply:
That's an easy enough change to make; however, it does open a potential class of bugs. 

As currently defined, `arm64-apple-ios-clang` will always be a clang install for ARM64 iOS devices. If we allow for `SDKROOT` as you describe, a user could easily think they're compiling an iOS build, but actually be using an iOS simulator SDK (or worse - a macOS or watchOS SDK).

This is especially problematic when the values are embedded in `sysconfig`, as the `CC` value embedded in sysconfig will become dependent on the end-user's environment.

You can still switch between different Xcode installs with `xcode-select` for testing purposes; I'd argue this is likely more useful for testing purposes, as Xcode leans heavily to having a single iOS SDK per Xcode install.

My inclination is that being explicit with `--sdk` in these scripts and overriding `SDKROOT` is preferable on balance, but I'll defer to your judgement if you feel strongly otherwise. If we *do* make this change, I'll add some docs clarifying how SDKROOT is interpreted, and clarifying that it's the user's responsibility to make sure SDKROOT matches their intended compilation target.

Reply:
I see two drawbacks with relying solely on `xcode-select`. One, `xcode-select -s` has effect system-wide, and so would affect all other uses and users of Xcode on that system. Of course, there is another way to effect this on a process basis: set a `DEVELOPER_DIR` env variable. But, two, even with that, AFAICT there still would be no way to select a non-default iOS SDK within a particular Xcode instance and, while it may not be the default case at the moment, it would be better to not preclude supporting more than one SDK available in an Xcode instance. I think the primary value of supporting all this would be during development and testing of Python iOS frameworks themselves and, as such, you likely wouldn't want to use them in an actual release of a Python iOS framework but there *might* be cases where you would. (And, the end-user of the framework, presumably someone developing an iOS app, would need to ensure they had a compatible developer environment anyway.)

So perhaps a slightly simpler way to address these concerns would be to define a new env variable that would allow overriding the SDK name in the stubs, something like `IOSSDK='iphoneos17.1'` (I'm not hung up on the name) instead of the default `iphoneos` ?, and mentioning setting `DEVELOPER_DIR` to use a non-default Xcode installation:

`xcrun --sdk ${IOSSDK:-iphoneos} clang -target arm64-apple-ios $@`

Or we could just defer this until a demonstrated need arises but I think it is worth considering.

Reply:
Passing in a specific SDK (or even better, a specific SDK *version*) with a custom environment variable makes sense to me. I'll push an update shortly that uses `IOS_SDK_VERSION` so you can get `iphoneosX.Y` and `iphonesimulatorX.Y` from a single environment variable (since the base names of the SDKs are stable, and I can't think of a use case where you'd want a different version for the device vs simulator SDK in a single build), along with docs covering the new variable and `DEVELOPER_DIR`